### PR TITLE
DB-8854 clear the corresponding nameTdCache entry after drop synonym/…

### PR DESCRIPTION
**Pull request by yxia92**
…alias.

### CODE REVIEW

**Comment by yxia92** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/DropAliasConstantOperation.java [hunk 1]
```diff
@@ -152,6 +152,11 @@ public static void invalidate(AliasDescriptor ad, DependencyManager dm, Language
         }
 
         dm.invalidateFor(ad, invalidationType, lcc);
+        if (ad.getAliasType() == AliasInfo.ALIAS_TYPE_SYNONYM_AS_CHAR) {
+            // invalidate nameTdCache
+            TableKey key = new TableKey(ad.getSchemaUUID(), ad.getName());
+            lcc.getDataDictionary().getDataDictionaryCache().nameTdCacheRemove(key);
```
Comment (dgomezferro) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/DropAliasConstantOperation.java]: I think this won't work for other RSs, would it?

Can you add a test for that case? Create alias in RS0, use it in RS1, drop it from RS0, see if you can recreate it from RS1

Comment (yxia92) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/DropAliasConstantOperation.java]: @dgomezferro , the fix works for other RSs. This function DropAliasConstantOperation.invalidate() is called by SpliceDatabase.registerDDL()->DDLUtils.preDropAlias(), so all region servers will invoke it.
I've pushed an IT to test it. Thanks!

Comment (dgomezferro) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/DropAliasConstantOperation.java]: I see, thanks!

**Pull request merged**